### DISCUSSION TITLE
Slow development time with TypeScript + SCSS Modules + SubComponents

### DISCUSSION THREAD

**Discussion by georgekrax** [vercel/next.js#24427] [Help]
During the last months, even if I use the `future: { weback5: true }` option in `next.config.js` file, I get very slow development time (~ 45+ seconds). I use TypeScript, and have enabled the `typescript: { transpileOnly: true  }` option too in the configuration file. Is there anyway SCSS and SCSS Modules to overload the compilation process. 

I added the `ProgressPlugin` and saw that most of the time used was for building the files, modules, and dependencies. Is there any way to minimize (or cache) and improve this process?
Also, I noticed that **Fast Refresh** will update and re-compile all the files of the SubComponents that I use, meaning that it will re-compile all the files of component's folder content.

Any ideas or suggestions to decrease the development time are welcome!

> What do you use at Vercel, where you state that the development process is really fast?!?

**Comment by bobarros** [comment]
It's hard to say what is slow not knowing all the details of your project and where you're running it.

Are you using Windows+WSL? Changing to WSL2 and also using Docker when viable improved performance 10x here.

The problem with WSL2 is that you need to start using **stored data only in the Linux filesystem**. Otherwise your workflow it's going to break.

But for a lot of o people is possible and also worth it.

**Reply by georgekrax** [reply]
Although, I have enabled WSL, have switched back to WSL 1 because it seemed more performant. I do not run `yarn dev` in the **WSL - Linux Ubuntu**  environment, but as regularly from the Windows CMD. I will switch back to WSL 2, and see if there is any improvement. 

Do you believe that SCSS Modules and SubComponents (e.g. `<Input.Password />`) can cause this because Fast Refresh will re-compile all the files in the folder tree structure?

**Reply by bobarros** [reply]
I'm not talking about executing from ubuntu with yarn dev or no. 

WSL 2 demands that you use Linux filesystem as your project directory. 

Linux containers only receive file change events if the original files are stored in the Linux filesystem. Speed performance it's only at its best this way, also.

So, you will not be able to use fast refresh if your files stay in Windows. In other words, that workflow it's messed up.

About your question regarding modules and components, I don't have an answer. I would have to investigate it.

**Reply by georgekrax** [reply]
I tried running it on WSL 2, but it still takes up to 5+ seconds to refresh, even after the update where they state that the reload time will range from 100ms to 200ms!

* Generally in **dev**  mode, it takes a lot of time to build a page, and navigate. In **production** this does not happen at all.
* My `/components` folder, seems to make this delay, but I am not entirely sure either. 
* It may be a custom package that I have created, but have not optimized and correctly packed for download via NPM and production use
